src/http/client.cc

```diff
@@ -307,31 +307,36 @@ future<> client::make_request(request req, reply_handler handle, abort_source& a
 }
 
 future<> client::do_make_request(request req, reply_handler handle, abort_source* as, std::optional<reply::status_type> expected) {
-    return do_with(std::move(req), std::move(handle), [this, as, expected] (request& req, reply_handler& handle) mutable {
-        return with_connection([this, &req, &handle, as, expected] (connection& con) {
-            return do_make_request(con, req, handle, as, expected);
-        }, as).handle_exception_type([this, &req, &handle, as, expected] (const std::system_error& ex) {
+    bool retry_with_new = false;
+    try {
+        co_await with_connection(coroutine::lambda([this, &req, &handle, as, expected] (connection& con) -> future<> {
+            co_await do_make_request(con, req, handle, as, expected);
+        }), as);
```

Comment:
probably we could add a `co_return` here? just for better readability. this helps reader understand that "if every goes fine, the princess and the ~~frog~~ prince lived happily ever after. FIN.".

Reply:
Hm...